packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -441,6 +442,41 @@ test('can specify custom test', () => {
   });
 });
 
+test('can specify vpc', () => {
```

Comment:
Can you also add the following unit tests:

- use `vpcSubnets`
- no `securityGroups` means one will be created for you
- can specify a `securityGroups` correctly
- vpc with no public security groups and no `allowPrivateSubnets` fails
- vpc with no public security groups and `allowPrivateSubnets` does not fail

Please also throw all these unit tests under a 

```ts
describe('canary in a vpc', () => {
```